lib/ansible/module_utils/basic.py

```diff
@@ -2369,9 +2369,10 @@ def fail_json(self, **kwargs):
             raise AssertionError("implementation error -- msg to explain the error is required")
         kwargs['failed'] = True
 
-        # add traceback if debug or high verbosity and it is missing
-        # Note: badly named as exception, it is really always been 'traceback'
-        if 'exception' not in kwargs and sys.exc_info()[2] and (self._debug or self._verbosity >= 3):
+        # Add traceback if debug or high verbosity and it is missing
+        # NOTE: Only do this for Python 3, as prior versions would add last (stack frame) exception
+        # NOTE: badly named as exception, it is really always been 'traceback'
+        if 'exception' not in kwargs and sys.version_info >= (3, 0, 0) and sys.exc_info()[2] and (self._debug or self._verbosity >= 3):
```

Comment:
don't we have PY3 var already to detect 3.x?

Reply:
It appears so :-) Fixed.